src/call/mod.rs

```diff
@@ -31,8 +31,58 @@ use crate::codec::{DeserializeFn, Marshaller, SerializeFn};
 use crate::error::{Error, Result};
 use crate::task::{self, BatchFuture, BatchType, CallTag, SpinLock};
 
-pub use crate::grpc_sys::grpc_status_code as RpcStatusCode;
-pub type RpcStatusCodeType = i32;
+/// An gRPC status code structure.
+/// This type contains constants for all gRPC status codes.
+#[derive(PartialEq, Clone, Copy, Debug)]
+pub struct RpcStatusCode(i32);
+
+impl From<i32> for RpcStatusCode {
+    fn from(code: i32) -> RpcStatusCode {
+        RpcStatusCode(code)
+    }
+}
+
+impl Into<i32> for RpcStatusCode {
+    fn into(self) -> i32 {
+        self.0
+    }
+}
+
+macro_rules! status_codes {
+    (
+        $(
+            ($num:expr, $konst:ident);
+        )+
+    ) => {
+        #[allow(dead_code)]
```

Comment:
Why do we need to `allow` here? Since this is all public api it should not be required.

Reply:
compiler hint, but don't see it this morning. Anyway, solved